### PR TITLE
Bump clock dependency a bit

### PR DESCRIPTION
**Pull request by k-bx**
Also add stack support.

Thank you!

### CODE REVIEW

**Comment by k-bx** [comment]
Will update PR to use submodules soon.

**Comment by k-bx** [comment]
Sorry, I don't actually see submodules used here. So I can either:

1. add submodules
2. remove stack support from here to only let users build it via main "snap" repo?

Please advice.

File: snap-server.cabal [hunk 1]
```diff
@@ -91,7 +91,7 @@ Library
     bytestring                          >= 0.9.1    && < 0.11,
     bytestring-builder                  >= 0.10.4   && < 0.11,
     case-insensitive                    >= 1.1      && < 1.3,
-    clock                               >= 0.4.1    && < 0.6,
+    clock                               >= 0.4.1    && < 0.7,
```
Comment (imalsogreg) [inline snap-server.cabal]: clock >= 0.6 breaks snap on ubuntu. We are waiting to hear back from clock's author on this: https://github.com/corsis/clock/issues/36

**Comment by mightybyte** [comment]
Unfortunately the new versions of clock are broken.  See this issue for more info:

https://github.com/corsis/clock/issues/36

I was actually thinking about doing the opposite and making the clock bounds smaller because clock-0.5.2 is also broken as well.  I am not inclined to widen the dependency until clock is fixed.

As far as submodules go, we had not set up submodules in this repo yet, although I guess it might not hurt.  If we do, we do not need to have heist as a submodule because snap-server does not depend on heist.

**Comment by hvr** [comment]
@mightybyte if you do set up submodules in this repo, it would be a good ideal to try to avoid git-submodule cycles between snap and snap-server, as otherwise `git clone --recursive` may have undesired behaviour...

File: snap-server.cabal (hunk 1 above)
Comment (k-bx) [inline snap-server.cabal]: @imalsogreg wow, these are very bad news, as stackage 4.1 has clock included into itself. I'll create a ticket to discuss this!

Comment (imalsogreg) [inline snap-server.cabal]: @k-bx Oh interesting :P Thanks for the info! I'd be curious to know what other libs depend on clock and experience problems.

Comment (k-bx) [inline snap-server.cabal]: not much it seems http://packdeps.haskellers.com/reverse/clock , formatting/tasty/amqp are of popular ones I guess

**Comment by mightybyte** [comment]
Closing since we now have constraints that depend on the new fixed version of clock.

**Pull request closed without merging**